File: sample_programs/cpp/attack/Attack.cpp
#include <iostream>
void init();
int ask(int game,int round,int*** record);
int min;       //Minimum usable value (inclusive)
int max;       //Maximum usable value (inclusive)
int gameNum;   //Number of games
int roundNum;  //Maximum number of rounds per game
int changeLimit; //Minimum number of times to change
std::string playerName = "playerCppAtk";
int ask(int game,int round,int*** record){
	return rand()%max+ min;
}
int main(void) {
	init();
	int ***record = new int**[gameNum];
	for(int i=0; i<gameNum; i++){
		record[i] = new int*[roundNum];
		for(int j=0; j<roundNum; j++){
			record[i][j] = new int[2];
		}
	}
	for(int i=0; i<gameNum; i++){
		for(int j=0; j<roundNum; j++){
			int askNum = ask(i,j,record);
			record[i][j][0] = askNum;
			std::cout << askNum << std::endl;
			int ud;
			std::cin >> ud;
			// 1 : if the predicted value is larger than the partner's number,
			//-1 : if it is smaller,
			// 0 : if it matches
			record[i][j][1] = ud;
			if(ud == 0) break;
		}
	}
	return 0;
}
void init() {
	std::cin >> min;
	std::cin >> max;
	std::cin >> gameNum;
	std::cin >> roundNum;
	std::cin >> changeLimit;
	std::cout << playerName << std::endl;
}

File: sample_programs/python/attack/Attack.py
import sys
import random

playerName = "playerAtk"
min = int(sys.stdin.readline())
max = int(sys.stdin.readline())
gameNum = int(sys.stdin.readline())
roundNum = int(sys.stdin.readline())
changeLimit = int(sys.stdin.readline())
print(playerName, flush=True)
record = [[[0 for k in range(2)] for i in range(roundNum)] for j in range(gameNum)]

def ask(game,round,record):
	return random.randrange(min,max+1)


for i in range(gameNum):
	for j in range(roundNum):
		askNum = ask(i,j,record)
		record[i][j][0] = askNum;
		print(askNum,flush=True)
		# 1 : if the predicted value is larger than the partner's number,
		#-1 : if it is smaller, 
		# 0 : if it matches
		ud = int(sys.stdin.readline())
		record[i][j][1] = ud;
		if ud == 0 :
			break